rust/lance-io/src/object_writer.rs

```diff
@@ -317,19 +323,36 @@ impl ObjectWriter {
             unreachable!()
         }
     }
+
+    pub async fn abort(&mut self) {
+        if let Some(abort_task) = self.inner_abort() {
+            abort_task.await;
+        }
+    }
+
+    fn inner_abort(&mut self) -> Option<BoxFuture<'static, ()>> {
+        let state = self.state.any_to_abort();
+        if let UploadState::InProgress { mut upload, .. } = state {
+            let abort_task = async move {
+                let _ = upload.abort().await;
+            }
+            .boxed();
+            Some(abort_task)
+        } else {
+            None
+        }
+    }
 }
 
 impl Drop for ObjectWriter {
     fn drop(&mut self) {
-        // If there is a multipart upload started but not finished, we should abort it.
-        if matches!(self.state, UploadState::InProgress { .. }) {
-            // Take ownership of the state.
-            let state =
-                std::mem::replace(&mut self.state, UploadState::Done(WriteResult::default()));
-            if let UploadState::InProgress { mut upload, .. } = state {
-                tokio::task::spawn(async move {
-                    let _ = upload.abort().await;
+        if let Some(abort_task) = self.inner_abort() {
+            if let Ok(handle) = Handle::try_current() {
+                handle.spawn(async move {
+                    abort_task.await;
                 });
+            } else {
+                warn!("Failed to abort object writer because runtime not found.");
```

Comment:
I don't think we need to warn here. The cleanup logic is best effort and we can just ignore it if failed.

Reply:
Makes sense to me, let me fix it.